### ISSUE TITLE
Deprecation warnings in Python 2.6 when running twisted.conch.test.test_filetransfer

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3897]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#3897|
|Type|defect|
|Created|2009-07-01 14:36:46Z|

`/home/jml/src/twisted/trunk/twisted/conch/ssh/filetransfer.py:834: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6`

```
twisted.conch.test.test_filetransfer
  TestConstants
    test_constantsAgainstSpec ...                                          [OK]
  TestFileTransferClose
    test_clientClosesChannelOnConnnection ...                              [OK]
    test_sessionClose ...                                                  [OK]
    test_stopConnectionServiceClosesChannel ...                            [OK]
  TestOurServerOurClient
    testClosedFileGetAttrs ...                                             [OK]
/home/jml/src/twisted/trunk/twisted/conch/ssh/filetransfer.py:834: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
    testDirectoryBad ...                                                   [OK]
/home/jml/src/twisted/trunk/twisted/conch/ssh/filetransfer.py:834: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
    testDirectoryCreation ...                                              [OK]
/home/jml/src/twisted/trunk/twisted/conch/ssh/filetransfer.py:834: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
    testExtendedRequest ...                                                [OK]
    testLinkDoesntExist ...                                                [OK]
/home/jml/src/twisted/trunk/twisted/conch/ssh/filetransfer.py:834: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
    testLinkPath ...                                                       [OK]
    testLinkSharesAttrs ...                                                [OK]
    testOpenDirectory ...                                                  [OK]
    testOpenFileAttributes ...                                             [OK]
    testOpenFileIO ...                                                     [OK]
    testOpenFileSetAttrs ...                                               [OK]
    testRemoveFile ...                                                     [OK]
/home/jml/src/twisted/trunk/twisted/conch/ssh/filetransfer.py:834: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
    testRenameFile ...                                                     [OK]
    testServerVersion ...                                                  [OK]
    test_openFileExtendedAttributes ...                                    [OK]
    test_openedDirectoryClosedWithConnection ...                           [OK]
    test_openedFileClosedWithConnection ...                                [OK]
```

Attachments:

* [filetransfer-deprecation-2687.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/cef/ceffdcb1aadf2115f6687fe784880a8428e0b6d8/6f5c6f60f2eb56997804a0099d8ee9bb18d5a129.diff) (970 bytes) - added by jml on 2009-07-01 14:54:14Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3897 3897
type__defect defect
reporter__jml jml
priority__normal normal
milestone__Python_2_6 Python-2.6
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1246459006000000 1246459006000000
changetime__1286996516000000 1286996516000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @jml<br>@exarkun set status to `new`|
|-|-|

Thanks.  We need more attention to things like this, or we'll never dig out from under the mountain of warnings presently emitted by the test suite. :)

The change here seems to look good.  The only thing I would say is missing is documentation about what the `message` attribute of `SFTPError` means/is/contains.  As far as I can tell, it's a string received over the network meant to explain a failure to a human.  Can you add something to this effect?  Then, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml set status to `closed`|
|-|-|

(In [27062]) Silence the deprecation warning in filetransfer.py that occurs in Python 2.6.

 * Author: jml
 * Reviewer: exarkun
 * Fixes [#3897](https://github.com/twisted/twisted/issues/3897)

`SFTPError` used to assign to 'message'. It's deprecated to assign to message
on Exceptions in Python 2.6. This patch changes `SFTPError` to use `_message`
internally, and makes the message attribute available as a property.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
